shell/renderer/preload_realm_context.cc

```diff
@@ -0,0 +1,295 @@
+  auto* controller = PreloadRealmLifetimeController::From(context);
+  if (controller)
+    return controller->GetContext();
+  return v8::MaybeLocal<v8::Context>();
+}
+
+electron::ServiceWorkerData* GetServiceWorkerData(
+    v8::Local<v8::Context> context) {
+  auto* controller = PreloadRealmLifetimeController::From(context);
+  return controller ? controller->service_worker_data() : nullptr;
+}
+
+void OnCreatePreloadableV8Context(
+    v8::Local<v8::Context> initiator_context,
+    electron::ServiceWorkerData* service_worker_data) {
+  v8::Isolate* isolate = initiator_context->GetIsolate();
+  blink::ScriptState* initiator_script_state =
+      blink::ScriptState::MaybeFrom(isolate, initiator_context);
+  DCHECK(initiator_script_state);
+  blink::ExecutionContext* initiator_execution_context =
+      blink::ExecutionContext::From(initiator_context);
+  DCHECK(initiator_execution_context);
+  blink::DOMWrapperWorld* world = blink::DOMWrapperWorld::Create(
+      isolate, blink::DOMWrapperWorld::WorldType::kShadowRealm);
+  CHECK(world);  // Not yet run out of the world id.
+
+  // Create a new ShadowRealmGlobalScope.
+  blink::ShadowRealmGlobalScope* shadow_realm_global_scope =
+      blink::MakeGarbageCollected<blink::ShadowRealmGlobalScope>(
+          initiator_execution_context);
+  const blink::WrapperTypeInfo* wrapper_type_info =
+      shadow_realm_global_scope->GetWrapperTypeInfo();
+
+  // Create a new v8::Context.
+  // Initialize V8 extensions before creating the context.
+  v8::ExtensionConfiguration extension_configuration =
+      blink::ScriptController::ExtensionsFor(shadow_realm_global_scope);
+
+  v8::Local<v8::ObjectTemplate> global_template =
+      wrapper_type_info->GetV8ClassTemplate(isolate, *world)
+          .As<v8::FunctionTemplate>()
+          ->InstanceTemplate();
+  v8::Local<v8::Object> global_proxy;  // Will request a new global proxy.
+  v8::Local<v8::Context> context =
+      v8::Context::New(isolate, &extension_configuration, global_template,
+                       global_proxy, v8::DeserializeInternalFieldsCallback(),
+                       initiator_execution_context->GetMicrotaskQueue());
+  context->UseDefaultSecurityToken();
+
+  // Associate the Blink object with the v8::Context.
+  blink::ScriptState* script_state =
+      blink::ScriptState::Create(context, world, shadow_realm_global_scope);
+
+  // Associate the Blink object with the v8::Objects.
+  global_proxy = context->Global();
+  blink::V8DOMWrapper::SetNativeInfo(isolate, global_proxy,
+                                     shadow_realm_global_scope);
+  v8::Local<v8::Object> global_object =
+      global_proxy->GetPrototype().As<v8::Object>();
+  blink::V8DOMWrapper::SetNativeInfo(isolate, global_object,
+                                     shadow_realm_global_scope);
+
+  // Install context-dependent properties.
+  std::ignore =
+      script_state->PerContextData()->ConstructorForType(wrapper_type_info);
```

Comment:
Are we just calling the function for the side effects here which installs properties and not needing to use the return value? Why do we need to use `std::ignore` here?

Reply:
This function was based on the implementation for creating a new shadow realm context here: https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/bindings/core/v8/shadow_realm_context.cc;l=104-106;drc=a45502c46d75f210c783e07384138379ea1e46e4

It _does_ compile without it, but I assume it's included to avoid chromium compiler warnings.